bundles/net.tourbook/src/net/tourbook/ui/views/nutrition/DialogSearchProduct.java

```diff
@@ -0,0 +1,1054 @@
+import net.tourbook.Messages;
+import net.tourbook.application.TourbookPlugin;
+import net.tourbook.common.UI;
+import net.tourbook.common.util.ColumnDefinition;
+import net.tourbook.common.util.ColumnManager;
+import net.tourbook.common.util.IContextMenuProvider;
+import net.tourbook.common.util.ITourViewer;
+import net.tourbook.common.util.PostSelectionProvider;
+import net.tourbook.common.util.TableColumnDefinition;
+import net.tourbook.data.TourData;
+import net.tourbook.data.TourNutritionProduct;
+import net.tourbook.nutrition.NutritionQuery;
+import net.tourbook.nutrition.NutritionUtils;
+import net.tourbook.nutrition.ProductSearchType;
+import net.tourbook.nutrition.openfoodfacts.Product;
+import net.tourbook.preferences.ITourbookPreferences;
+import net.tourbook.tour.TourManager;
+import net.tourbook.web.WEB;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.action.IMenuManager;
+import org.eclipse.jface.action.MenuManager;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.IDialogSettings;
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.layout.PixelConverter;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.viewers.CellLabelProvider;
+import org.eclipse.jface.viewers.ColumnViewer;
+import org.eclipse.jface.viewers.ComboViewer;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredContentProvider;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.jface.viewers.ViewerCell;
+import org.eclipse.jface.viewers.ViewerComparator;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Listener;
+import org.eclipse.swt.widgets.Menu;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.TableColumn;
+import org.eclipse.swt.widgets.ToolTip;
+import org.eclipse.swt.widgets.Widget;
+
+public class DialogSearchProduct extends TitleAreaDialog implements ITourViewer, PropertyChangeListener {
```

Comment:
I had requested this feature...

![image](https://github.com/mytourbook/mytourbook/assets/1283445/f026246f-0f95-4a44-995b-46daa9b475dc)

...for the search dialog but you have applied it to the tour nutrition view

Can you also apply it for the viewer in the search dialog

Reply:
I had not put it to the search dialog because it was not working as expected. 
But now I have put it to the search dialog (even though it is still not perfect)

Reply:
>(even though it is still not perfect)

What is not perfect?

I discovered these issues in the search dialog

* After columns are customized, then the viewer is empty (the viewer control was recreated)
* The state for customized columns are not saved, when the search dialog is reopened, then the default columns are displayed again

Reply:
>What is not perfect?

You can still see the "hack" column if you look closely and therefore it's not very "clean"

![image](https://github.com/mytourbook/mytourbook/assets/25235665/73ed7463-415f-4f7a-97fb-90e129daaae5)


 >   After columns are customized, then the viewer is empty (the viewer control was recreated)
>    The state for customized columns are not saved, when the search dialog is reopened, then the default columns are displayed again

This should be fixed now

Reply:
I was wondering why the column width was 7, now I found it

![image](https://github.com/mytourbook/mytourbook/assets/1283445/9bdf6b6c-b937-4524-97fa-ab9e1979a198)

For such an "invisible" column I will (sometimes) adjust the column manager to allow column widths of 0

Reply:
It's now almost 16 years ago

![image](https://github.com/mytourbook/mytourbook/assets/1283445/f8e59709-e7df-4627-8820-4e831ef9804b)

Reply:
16 years is a long time ago ! Before you put the code on Github ?

Reply:
> Before you put the code on Github ?

Yes

![image](https://github.com/mytourbook/mytourbook/assets/1283445/92fce65e-01be-4d36-b065-683d1e1531ac)